isis/src/qisis/objs/ControlPointEditWidget/ControlPointEditWidget.cpp

```diff
@@ -2386,6 +2409,59 @@ namespace Isis {
   }
 
 
+  /**
+  * Add registration TemplateList to combobox when imported to project
+  *
+  * @param templateList Reference to TemplateList that was imported
+  */
+  void ControlPointEditWidget::addTemplates(TemplateList *templateList) {
+    if(templateList->type() == "registrations") {
+      for(int i = 0; i < templateList->size(); i++) {
+        m_templateComboBox->addItem(templateList->at(i)->importName()
+                                    + "/" + FileName(templateList->at(i)->fileName()).name());
+      }
+    }
+  }
+  
+  
+  /**
+  * Appends the filename to the registrations path (unless this is the default template) and calls 
+  * setTemplateFile for the control measure
+  *
+  * @param filename This is the import directory and the filename of the template file selected from
+  *                 the QComboBox.
+  */
+  void ControlPointEditWidget::setTemplateFile(QString filename) {
+    QString expandedFileName = filename;
+    if(!filename.startsWith("$base")){
```

Comment:
Will the template file ever be in a different area non-project area like `$hayabusa`?

Reply:
Not as of right now. The default template file in $base is hard-coded. The only other files that will exist as an option in this combobox or imported files. If other files are added in the future, this logic will need to change. Do you think this is reasonable now, or should I find another way of distinguishing?

Reply:
No, you don't need to change anything. If `$base` is already hard coded for the template file area, the rest of this makes sense, and I think that it's okay that if this changes in the future, the other logic will need to be update too.